apps/omg_watcher_info/lib/omg_watcher_info/api/stats.ex

```diff
@@ -20,49 +20,58 @@ defmodule OMG.WatcherInfo.API.Stats do
   alias OMG.WatcherInfo.DB.Block
   alias OMG.WatcherInfo.DB.Transaction
 
+  @seconds_in_twenty_four_hours 86_400
+
   @doc """
   Retrieves network statistics.
   """
   def get() do
-    timestamps_all_time = Block.get_timestamps()
-    timestamps_last_24_hours = Block.get_timestamps_last_24_hours()
+    end_datetime = DateTime.to_unix(DateTime.utc_now())
+    start_datetime_24_hours = end_datetime - @seconds_in_twenty_four_hours
+
+    timestamps_all_time = Block.all_timestamps()
+    timestamps_24_hours = Block.all_timestamps_between(start_datetime_24_hours, end_datetime)
 
     response = %{
       transaction_count: %{
-        all_time: Transaction.get_count(),
-        last_24_hours: Transaction.get_count_last_24_hour()
+        all_time: Transaction.count_all(),
+        last_24_hours: Transaction.count_all_timestamp_between(start_datetime_24_hours, end_datetime)
       },
       block_count: %{
-        all_time: Block.get_count(),
-        last_24_hours: Block.get_count_last_24_hour()
+        all_time: Block.count_all(),
+        last_24_hours: Block.count_all_timestamp_between(start_datetime_24_hours, end_datetime)
       },
-      average_block_interval: %{
+      average_block_interval_seconds: %{
         all_time: get_average_block_interval(timestamps_all_time),
-        last_24_hours: get_average_block_interval(timestamps_last_24_hours)
+        last_24_hours: get_average_block_interval(timestamps_24_hours)
       }
     }
 
     {:ok, response}
   end
 
-  @spec get_average_block_interval([%{timestamp: integer}]) :: float | String.t()
+  @doc """
+  Calculates the average of the differences between timestamps.
+  """
+  @spec get_average_block_interval([%{timestamp: non_neg_integer()}]) :: float | String.t()
   def get_average_block_interval(timestamps) do
-    case length(timestamps) do
-      n when n < 2 ->
-        String.to_atom("N/A")
+    case timestamps do
+      [_, _ | _] ->
+        first =
+          timestamps
+          |> List.first()
+          |> Map.get(:timestamp)
 
-      n when n >= 2 ->
-        timestamps
-        |> Enum.chunk_every(2, 1, :discard)
-        |> Enum.map(fn [a, b] -> b.timestamp - a.timestamp end)
-        |> average()
-    end
-  end
+        last =
+          timestamps
+          |> List.last()
+          |> Map.get(:timestamp)
 
-  @spec average([integer]) :: float
-  def average(diff_array) do
-    diff_array
-    |> Enum.reduce(0, fn current, acc -> current + acc end)
-    |> Kernel./(length(diff_array))
+        # Formula for average of difference
+        (last - first) / (length(timestamps) - 1)
```

Comment:
hm... so if, by coincidence, last item and first item have accidental huge difference (network latency, postgres latency)... this wouldn't be representable or am I wrong?

Reply:
you're wrong @InoMurko